aten/src/ATen/native/cudnn/BatchNorm.cpp

```diff
@@ -88,14 +88,21 @@ std::tuple<Tensor, Tensor, Tensor, Tensor> cudnn_batch_norm(
   }
 
   cudnnBatchNormMode_t mode;
+  at::MemoryFormat memory_format = input->suggest_memory_format();
```

Comment:
can you please factor out this format selection function? Otherwise it can go out of sync in forward and backward.